src/libraries/System.Composition.TypedParts/src/System/Composition/Hosting/InstanceExportDescriptorProvider.cs

```diff
@@ -0,0 +1,26 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Composition.Hosting.Core;
+
+namespace System.Composition.Hosting
+{
+    internal class InstanceExportDescriptorProvider : SinglePartExportDescriptorProvider
```

Comment:
>InstanceExportDescriptorProvider [](http://example.com/codeflow?start=19&length=32)

Is it possible to get rid of SinglePartExportDescriptorProvider and have InstanceExportDescriptorProvider simply wrap a CompositionContract and _exportedInstance objects?

Would that be simpler?

Reply:
I simply copied them from other implementations (the one used in arcade). I haven't looked around for other usages.

Reply:
Oh, I didn't know you were copying this from another repo. I would keep this implementation then as this should be used and validated there.